jetty-websocket/websocket-javax-server/src/main/java/org/eclipse/jetty/websocket/javax/server/internal/JavaxWebSocketServerContainer.java

```diff
@@ -301,4 +303,19 @@ protected void doStart() throws Exception
             deferredEndpointConfigs.clear();
         }
     }
+
+    @Override
+    public CompletableFuture<Void> shutdown()
+    {
+        LifeCycle.stop(sessionTracker);
+        CompletableFuture<Void> shutdown = new CompletableFuture<>();
+        shutdown.complete(null);
+        return shutdown;
+    }
```

Comment:
Try something like the below.  This lets the caller wait for this to complete or not, as it likes.

```suggestion
    {
        CompletableFuture<Void> shutdown = new CompletableFuture<>();
        new Thread(()->
        {
            try
            {
                LifeCycle.stop(sessionTracker);
            }
            catch(Exception e)
            {
                LOG.warn(e);
            }
            finally
            {
                shutdown.complete(null);
            }
        }.start();
        return shutdown;
    }
```

Reply:
Shouldn't the exception be caught and cause the `shutdown` `CompletableFuture` to fail with an exception?